src/superphot_plus/fit_numpyro.py

```diff
@@ -100,18 +103,27 @@ def run_mcmc(lc, sampler="NUTS", t0_lim=None, plot=False):
     rng_key = random.PRNGKey(4)
     rng_key, rng_key_ = random.split(rng_key)  # pylint: disable=unused-variable
 
-    ref_band_idx = 1  # red band # pylint: disable=unused-variable
-    n_params = 14  # pylint: disable=unused-variable
-
+    all_priors = priors.to_numpy().T
+    ref_band = priors.reference_band
+        
+    n_params = len(all_priors.T)
+    unique_bands = priors.ordered_bands
+    ref_band_idx = np.argmax(unique_bands == ref_band)
+
+    # Require data in both the g and r bands.
+    for ub in unique_bands:
+        if lc.obs_count(ub) == 0:
+            return None
+        
     tdata = lc.times
     fdata = lc.fluxes
     ferrdata = lc.flux_errors
-    bdata = lc.band_as_int(["g", "r"])  # change to integers
-
-    max_flux = np.max(fdata[PAD_SIZE:] - np.abs(ferrdata[PAD_SIZE:]))
-    inc_band_ix = np.arange(0, PAD_SIZE)
+    
+    ref_padded_idx = ref_band_idx*PAD_SIZE
+    max_flux, max_flux_time = lc.find_max_flux()
```

Comment:
To be consistent with the previous code this should be `lc.find_max_flux(band=ref_band)`

Otherwise it will find the max flux over all bands.